### DISCUSSION TITLE
Import the Exceptions in the unit tests

### DISCUSSION THREAD

**Discussion by Sysdraw** [directus/directus#13446] [Feature Requests]
Hello everyone,

I'm sorry my english is bad, but I will do my best 👍 

I'm creating a unit test with Jest and I have a problem, how I can import `InvalidPayloadException` ?

```javascript
test("Bad authentification", async () => {
    process.env.CLIENT_ID = "Bad CLIENT_ID";
    try {
        await getToken();
    } catch (error) {
        expect(error.message).toThrow(
            new InvalidPayloadException("InvalidPayloadException is not defined")
        )
    }
});
```

@br41nslug gave me an "solution", use `BaseException`  :
```javascript
import { BaseException } from '@directus/shared/exceptions';

test("Bad authentification", async () => {
    process.env.CLIENT_ID = "Bad CLIENT_ID";
    try {
        await getToken();
    } catch (error) {
        expect(error.message).toThrow(
            new BaseException("InvalidPayloadException is not defined")
        )
    }
});
```

I would like an improvement/feature about the exceptions, but don't worry it's not a priority.

**Comment by rijkvanzanten** [comment]
Is this a unit test for a hook/endpoint?

**Reply by Sysdraw** [reply]
In my situation for a Hook. But we can have the same problem for a endpoint.

**Comment by br41nslug** [comment]
For these sorts of testing use cases i think we should consider moving these exceptions exposed to extension to the `@directus/shared` package

**Comment by rijkvanzanten** [comment]
Heya! Thank you for taking the time to submit this request! 

It has been over **90 days**, and this discussion has not received at least **15 votes** from the community. This means that we don't feel like there's enough community interest to warrant further R&D into this topic at this time. :ice_cube:

This request will now be closed to keep our discussions tidy. Please reach out if you have any questions!

_For more information, see our [Feature Request Process](https://docs.directus.io/contributing/feature-request-process.html)._
